bots/guess_bot.js

```diff
@@ -0,0 +1,65 @@
+require('dotenv').config({silent: true})
+
+var Connection = require('../connection')
+
+const startWord = 'start'
+const guessWord = 'guess'
+const beginWord = 'begin'
+const rewardWord = 'reward'
+
+const startRE = new RegExp(startWord)
+const guessRE = new RegExp(guessWord)
+const beginRE = new RegExp(beginWord)
+const rewardRE = new RegExp(rewardWord)
+
+class GuessBot {
+  constructor() {
+    this.SLACK_CHANNEL = process.env.SLACK_CHANNEL
+    this.connection = new Connection()
+    this.connection.listen({
+      channel: this.SLACK_CHANNEL,
+      callback: this._handleMessage.bind(this),
+    })
+    this.connection.start().then(this._setId)
+  }
+
+  _setId = (id) => {
+    this.id = id
+    this.myselfRE = new RegExp(id)
+  }
+
+  _handleBegin(text, godBotId) {
+    const parts = text.split(beginRE)
+    const actionSpace = parts[1].trim().split(', ')
+    this.handleExplorationSpace(actionSpace, godBotId)
+  }
+
+  _handleReward(text) {
+    const parts = text.split(rewardRE)
+    const [reward, guess, totalScore, remaining] = parts[1].trim().split(', ')
+    this.handleReward(reward, guess, totalScore, remaining)
```

Comment:
What do you think about parsing the reward, guess, and remaining values as numbers before passing it into `handleReward`? I think that would make custom bots just a tiny bit easier to implement.

Reply:
Addressed in 155adf92aa634dcb98a1fa7932fa7e38183449b6

In testing, this may lead to some NaN values if the implementer accidentally forgets commas.